src/components/Loader/Loader.module.css

```diff
@@ -1,6 +1,12 @@
 @import '../../stylesheet/vars.css';
-.test {
-  width: 100px;
-  height: 100px;
-  background-color: var(--background-primary);
+.loader {
+  z-index: 100;
+  position: fixed;
+  top: 0;
+  width: 100vw;
+  height: 100vh;
+  display: flex;
+  align-items: center;
+  justify-content: center;
+  background-color: var(--color-brand-secondary);
```

Comment:
Zamiast tła jednokolorowego ustaw np:
backdrop-filter: blur(10px);
wtedy wszystko pod spodem będzie się blurować fajnie- możesz też poczytać o backdrop-filter i wybrać ten który Twoim zdaniem najlepiej pasuje -
https://developer.mozilla.org/en-US/docs/Web/CSS/backdrop-filter
 jak tu ostawisz to już będzie gotowy komponent na wszystkie breakpointy: